claims.go

```diff
@@ -269,8 +269,37 @@ func (c *Claims) nonceByProfile(profile string) ([]byte, error) {
 	return nonce, nil
 }
 
+func (c *Claims) SetNonce(nonce []byte) error {
+	if err := isPSAHashType(nonce); err != nil {
+		return err
+	}
+
+	profile, err := c.GetProfile()
+	if err != nil {
+		return err
+	}
+
+	switch profile {
+	case PSA_PROFILE_1:
+		c.LegacyNonce = &nonce
+	case PSA_PROFILE_2:
+		if c.Nonce != nil {
+			return errors.New("nonce already set")
```

Comment:
Why overwriting of nonce not permitted ? Any specific reason for not allowing an overwrite?

User should have a chance to correct his or her mistake simply?

Reply:
That's a rule that we establish to simplify the flow.

Besides, I'm not sure how one can correct a mistake in this case?